base/string.jl

```diff
@@ -185,10 +185,9 @@ function search(s::AbstractString, c::Chars, i::Integer)
         return 1 <= i <= nextind(s,endof(s)) ? i :
                throw(BoundsError(s, i))
     end
-    if i < 1
+    if i < 1 || i > nextind(s,endof(s))
```

Comment:
Doesn't look like you're catching all invalid indices here, are you? Shouldn't this be `i > endof(s)`?
```julia
julia> s = "café"
"café"

julia> nextind(s,endof(s))
6

julia> s[5]
ERROR: ArgumentError: invalid UTF-8 character index
 in next at ./utf8.jl:80
 in getindex at string.jl:62
```